File: src/main/java/eotree/EOMeta.java
package eotree;

/**
 * EBNF representation:
 * <code>
 * '+' name '␣' ANY EOL
 * </code>
 */
public abstract class EOMeta extends EONode {
    public String name;
    public String value;

    public EOMeta(String name, String value) {
        this.name = name;
        this.value = value;
    }

    @Override
    public String generateEO(int indent) {
        return "+" + name + " " + value;
    }
}

File: src/main/java/eotree/EOMeta.kt
package eotree

/**
 * EBNF representation:
 * `
 * '+' name ' ' ANY EOL
` *
 */
class EOMeta(var name: String, var value: String) : EONode() {
    override fun generateEO(indent: Int): String =
        "+$name $value"

    override fun toString(): String = "+$name $value"
}